### PR TITLE
provide log for test suite run incl log lines before/after/between te…

### PR DESCRIPTION
**Pull request by jjelschen**
…st runs

### CODE REVIEW

File: org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/loglines/ScanningLogFinder.xtend [hunk 1]
```diff
@@ -24,69 +27,101 @@ import static extension java.nio.file.Files.readAllLines
...
-	
-	override getLogLinesForTestStep(TestExecutionKey key) {
-		return getLogLinesForTestStep(key, LogLevel.TRACE)
-	}
 
-	override getLogLinesForTestStep(TestExecutionKey key, LogLevel logLevel) {
-		Validate.notBlank(key?.suiteId, ILLEGAL_TEST_EXECUTION_KEY_MESSAGE, key?.toString)
-		Validate.notBlank(key?.suiteRunId, ILLEGAL_TEST_EXECUTION_KEY_MESSAGE, key?.toString)
+	@FinalFieldsConstructor
+	static class LogLineSelector {
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/loglines/ScanningLogFinder.xtend]: FullLogLineSelector

File: org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/loglines/ScanningLogFinder.xtend [hunk 2]
```diff
@@ -24,69 +27,101 @@ import static extension java.nio.file.Files.readAllLines
...
+
+		protected val TestExecutionKey key
+		protected val extension WorkspaceProvider
+
+		def String[] getRelevantLines() {
+			return key.logFile.lines(UTF_8).findStart.findEnd.collect(Collectors.toList)
+		}
+
+		protected def Stream<String> findStart(Stream<String> lines) { return lines }
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/loglines/ScanningLogFinder.xtend]: /** default behaviour does not limit the start */

File: org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/loglines/ScanningLogFinder.xtend [hunk 3]
```diff
@@ -24,69 +27,101 @@ import static extension java.nio.file.Files.readAllLines
...
+
+		protected def Stream<String> findEnd(Stream<String> lines) { return lines }
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/loglines/ScanningLogFinder.xtend]: /** default behaviour does not limit the end */

File: org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/loglines/ScanningLogFinder.xtend [hunk 4]
```diff
@@ -24,69 +27,101 @@ import static extension java.nio.file.Files.readAllLines
...
+	static class TestStepLogLineSelector extends LogLineSelector {
 
-		val matcher = FileSystems.^default.getPathMatcher('''glob:testrun.«key.suiteId»-«key.suiteRunId»--.*.log''')
-		val logFile = workspace.toPath.resolve('logs').list //
-		.filter[matcher.matches(fileName)] //
-		.findFirst //
-		.orElseThrow [
-			new FileNotFoundException('''No log file for test execution key '«key?.toString»' found.''')
-		]
+		val Pattern enterPattern
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/loglines/ScanningLogFinder.xtend]: rename to compiledEnterPattern?
rename to compiledLeavePattern?

File: org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/loglines/ScanningLogFinder.xtend [hunk 5]
```diff
@@ -24,69 +27,101 @@ import static extension java.nio.file.Files.readAllLines
...
-		if (callTreeId.nullOrEmpty) {
-			callTreeId = ROOT_ID
+	private def getLogLineSelector(TestExecutionKey key) {
+		return if (key.caseRunId.nullOrEmpty) {
+			new LogLineSelector(key, workspaceProvider)
+		} else if (key.callTreeId.nullOrEmpty) {
+			new TestStepLogLineSelector(key,
+				workspaceProvider, '''@TESTRUN:ENTER:«key.suiteId».«key.suiteRunId».«key.caseRunId»''', '''@TESTRUN:LEAVE:«key.suiteId».«key.suiteRunId».«key.caseRunId»''')
+		} else {
+			new TestStepLogLineSelector(key,
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/loglines/ScanningLogFinder.xtend]: callTreeId is unique only within one test run (cascade log line selectors?)

**Review by gunther-bachmann — CHANGES_REQUESTED** [review]

File: org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/loglines/ScanningLogFinder.xtend (hunk 2 above)
Comment (jjelschen) [reply org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/loglines/ScanningLogFinder.xtend]: (but I'm going with AE _behavior_ :smile: )

**Review by gunther-bachmann — APPROVED** [review]

**Pull request merged**